### ISSUE TITLE
Demo should not depend on URL rewriting

### ISSUE THREAD

**Issue by joostdecock** [joostdecock/core#34]
Currently, the demo depends on rewriting URLs (based on the mod_rewrite rules in htaccess).
That's an extra requirement for people who want to kick the tires, and one that doesn't really provide much value (apart from slightly cleaner URLs).

The demo should be adapted so that it doesn't use rewritten URLs, or (best case) that this behaviour can be configured, but is disabled by default.

**Comment by straytaoist** [comment]
Well, those are some of the hoops I kinda went through (and hence my README update).

But yeah, this should kinda be sorted. I might play with a proper dispatcher (I am not a fan of the way this is done, but I blame PHP for that) in a bit. And that would give a nice clean, definable API too...

*edit* That probably sounds snippier than I meant it! I guess I've been writing APIs in golang for the past bit, and they are lovely.

**Comment by cabi** [comment]
@straytaoist what as example mean you with proper dispatcher? A central routing for the API functionality? Have you a example from golang? I love good code from other languages. :)

Maybe we find a nice way to make some API things smoother.

**Comment by joostdecock** [comment]
I have closed this issue with a2895f80a7fc743074391bbcf4e37a95303b2884. 

This is is no way to discourage this dispatcher business. However, this specific issue deals with the fact that the demo should not depend on URL rewriting. And given that my commit fixes that, I closed it.

I'm going to create a new issue for this dispatcher story, and I would encourage @straytaoist to outline his ideas so we can see how they can be implemented.
